src/spikeinterface/sortingcomponents/waveforms/waveform_thresholder.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+import json
+from typing import List, Optional
+import scipy.signal
+import numpy as np
+
+from spikeinterface.core import BaseRecording
+from spikeinterface.sortingcomponents.peak_pipeline import (
+    PipelineNode,
+    WaveformExtractorNode,
+)
+
+
+class WaveformThresholder(WaveformExtractorNode):
+    """
+    Waveform Thresholder based on selected feature
+    This allow to perform an adaptive masking, by setting to 0 channels
+    that have a given feature below a given threshold
+
+    Parameters
+    ----------
+    feature: string in ['ptp', 'mean', 'energy', 'v_peak']
+        the feature that should be considered
+    threshold: float
+        the threshold below which channels are set to 0
+    """
+
+    def __init__(
+        self, recording: BaseRecording, return_output: bool = True, 
+        parents: Optional[List[PipelineNode]] = None,
+        feature = 'ptp',
```

Comment:
```suggestion
        feature: Literal['ptp', 'mean', 'energy', 'v_peak'] = 'ptp'
```
You need to do from typing import Literal